src/client/mod.rs

```diff
@@ -415,6 +445,29 @@ impl Client {
         }
     }
 
+    pub(crate) fn build_watch(
```

Comment:
I think it would make more sense to move some of this work to `Aggregate`, i.e. with a `new_watch` function that does the work currently in this function to build the aggregation pipeline and create a new `Aggregate`. That way we could probably eliminate this method as initializing the `ChangeStreamData` is just one line, and the `watch` methods would look something like this:

```
// resolve options
let aggregate = Aggregate::new_watch(target, pipeline, options)?;
let cursor = client.execute_operation(aggregate)?;
Ok(ChangeStream::new(
    cursor,
    ChangeStreamData::new(pipeline, client, target, options),
))
```
In general I think it would be a bit more clear to associate the work in this function with the `Aggregate` struct rather than the `Client` struct. What do you think?

Reply:
Good thought, done.